pkg/engine/internal/types/operators.go

```diff
@@ -109,3 +114,29 @@ func (t BinaryOp) String() string {
 		panic(fmt.Sprintf("unknown binary operator %d", t))
 	}
 }
+
+// UnwrapOp denotes the kind of [UnwrapOp] operation to perform.
+type UnwrapOp uint32
+
+// Recognized values of [UnwrapOp].
+const (
+	// UnwrapOpInvalid indicates an invalid unwrap operation.
+	UnwrapOpInvalid UnwrapOp = iota
+	Unwrap
```

Comment:
Is this considered an `unwrap INT` or `unwrap FLOAT`, or generic `unwrap NUMBER`?

Reply:
renamed to cast and got made the base case more explicit by naming it `CastFloat`